// my-library/dist/index.js
// src/my-library.ts
import { Client, Collection } from "discord.js";
import { createAudioPlayer } from "@discordjs/voice";
var Cooldown = class {
  storage;
  constructor() {
    this.storage = {};
  }
  delete(key) {
    const keyString = String(key);
    if (this.storage[keyString]) {
      delete this.storage[keyString];
      return true;
    }
    return false;
  }
  has(key) {
    const keyString = String(key);
    return this.storage[keyString] !== void 0;
  }
  isExpired(key) {
    const keyString = String(key);
    const n = Date.now();
    if (this.storage[keyString] < n || !this.storage[keyString]) {
      return true;
    }
    return false;
  }
  set(key, value) {
    const keyString = String(key);
    this.storage[keyString] = value;
  }
};
var MyClient = class extends Client {
  commands;
  apps;
  contextMenus;
  modals;
  selectMenus;
  buttons;
  guild;
  queue;
  player;
  constructor(options) {
    super(options);
    this.commands = new Collection();
    this.apps = [];
    this.contextMenus = new Collection();
    this.modals = new Collection();
    this.selectMenus = new Collection();
    this.buttons = new Collection();
    this.queue = [];
  }
  setGuild(guild) {
    this.guild = guild;
  }
  enableAudioPlayer() {
    this.player = createAudioPlayer();
  }
};
export {
  Cooldown,
  MyClient
};
