// contents/dist/abmeldung-BMCGIIZL.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/Staff/abmeldung.ts
init_esm_shims();
import { ButtonBuilder, ActionRowBuilder, EmbedBuilder } from "discord.js";
var abmeldung_default = {
  id: "abmeldung",
  async execute(interaction) {
    const reason = interaction.fields.getTextInputValue("reason");
    const zeitraum = interaction.fields.getTextInputValue("zeitraum");
    const bemerkungen = interaction.fields?.getTextInputValue("bemerkungen");
    const channel = interaction.guild.channels.cache.get("1201604935943475250");
    const embed = new EmbedBuilder({
      title: "Antrag auf Abmeldung",
      fields: [
        { name: "Antragsteller", value: `${interaction.member}` },
        { name: "Grund", value: reason },
        { name: "Zeitraum", value: zeitraum },
        { name: "Bemerkungen", value: bemerkungen || "*Keine weiteren Anmerkungen*" }
      ],
      timestamp: /* @__PURE__ */ new Date(),
      color: 15083313
    });
    const row = new ActionRowBuilder().addComponents([
      new ButtonBuilder({
        label: "Akzeptieren",
        style: 3,
        customId: "accept",
        emoji: "\u2705"
      }),
      new ButtonBuilder({
        label: "Ablehnen",
        style: 4,
        customId: "decline",
        emoji: "\u{1F4DB}"
      })
    ]);
    const ping = "<@&1170957646942191688> <@&1181259236408311828>";
    await channel.send({ content: ping, embeds: [embed], components: [row] });
    await interaction.reply({ content: "Dein Antrag auf Abmeldung wurde \xFCbermittelt!", ephemeral: true });
  }
};
export {
  abmeldung_default as default
};
