// contents/dist/accept-GN2Q5XEM.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Buttons/Abmeldung/accept.ts
init_esm_shims();
import { ModalBuilder, TextInputBuilder, TextInputStyle, ActionRowBuilder } from "discord.js";
var allowed = [
  "773072144304963624",
  "1196808890227118201",
  "1035799257409667102"
];
var accept_default = {
  id: "accept",
  async execute(interaction) {
    if (allowed.some((id) => interaction.user.id == id)) {
      const modal = new ModalBuilder({
        title: "Abmeldung akzeptieren",
        customId: "accept"
      });
      const reason = new TextInputBuilder({
        label: "Grund (optional)",
        customId: "accept_reason",
        required: false,
        style: TextInputStyle.Paragraph,
        maxLength: 1e3
      });
      const row1 = new ActionRowBuilder().addComponents(reason);
      modal.addComponents(row1);
      await interaction.showModal(modal);
    } else return interaction.reply({ content: "Du darfst keine Abmeldungen akzeptieren", ephemeral: true });
  }
};
export {
  accept_default as default
};
