// contents/dist/anti-selfbot-6JDOXWKL.js
import {
  unbanRequestButton
} from "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Events/Security/anti-selfbot.ts
init_esm_shims();
import { Colors, EmbedBuilder, Events } from "discord.js";
var anti_selfbot_default = {
  name: Events.MessageCreate,
  async execute(message) {
    if (message.author.bot) return;
    if (message.member?.roles.cache.has("1156298949301379212" /* staff */)) return;
    const { content } = message;
    if (!message.member?.roles.cache.has("1156298949301379212" /* staff */) && content.includes("@everyone") && content.includes("discord.gg/")) {
      await message.delete();
      await message.member?.ban({ deleteMessageSeconds: 3600, reason: "Sent discord invite and pinged everyone" }).catch(async (err) => {
        await message.member?.timeout(3600, "Sent discord invite and pinged everyone");
      });
      const embed = new EmbedBuilder({
        title: "Gebannt",
        description: "Du wurdest automatisch gebannt, da du einen Invite geschickt hast und @everyone gepingt hast.\n\nDu kannst mit dem Button unten einen Entbannungsantrag stellen.",
        color: Colors.DarkRed,
        timestamp: /* @__PURE__ */ new Date(),
        footer: { text: "Sicherheitsbann" }
      });
      await message.member.send({ embeds: [embed], components: [unbanRequestButton] }).catch((err) => {
        console.log(err);
      });
    }
  }
};
export {
  anti_selfbot_default as default
};
