// contents/dist/apply-IPSJK42P.js
import {
  ticketButtons
} from "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/Ticket/apply.ts
init_esm_shims();
import { EmbedBuilder, ChannelType } from "discord.js";
var apply_default = {
  id: "apply",
  async execute(interaction) {
    const channel = await interaction.guild.channels.create({
      name: `ticket-${interaction.user.username}`,
      type: ChannelType.GuildText,
      parent: "1173314530521129042" /* ticket */,
      permissionOverwrites: [
        { id: interaction.user, allow: ["SendMessages", "ViewChannel"] },
        { id: interaction.guild.roles.everyone, deny: ["ViewChannel"] }
      ]
    });
    const job = interaction.fields.getTextInputValue("job");
    const experience = interaction.fields.getTextInputValue("experience");
    const name = interaction.fields.getTextInputValue("name");
    const age = interaction.fields.getTextInputValue("age");
    const onlinetime = interaction.fields.getTextInputValue("onlinetime");
    const embed1 = new EmbedBuilder({
      description: "Dies ist ein Bewerbungs Ticket!\nBitte habe etwas Geduld, ein Teammitglied wird sich demn\xE4chst um deine Bewerbung k\xFCmmern",
      color: 255
    });
    const embed2 = new EmbedBuilder({
      fields: [
        { name: "Als was m\xF6chtest du dich bewerben?", value: job },
        { name: "Erfahrungen", value: experience },
        { name: "Name", value: name },
        { name: "Alter", value: age },
        { name: "Onlinezeiten", value: onlinetime }
      ],
      color: 255
    });
    await channel.send({ content: "@everyone", embeds: [embed1, embed2], components: [ticketButtons] }).then(async (message) => {
      await message.edit({ content: "" });
      interaction.reply({ content: `Dein Ticket wurde erstellt: ${channel}`, ephemeral: true });
    });
  }
};
export {
  apply_default as default
};
