// contents/dist/ce_description-SCOMUBEG.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Buttons/CustomEmbed/ce_description.ts
init_esm_shims();
import { ModalBuilder, TextInputBuilder, ActionRowBuilder } from "discord.js";
var ce_description_default = {
  id: "ce_description",
  async execute(interaction) {
    if (interaction.user.username != interaction.message.embeds[0].author.name) return interaction.reply({ content: "Du darfst an diesem Embed nichts \xE4ndern", ephemeral: true });
    const modal = new ModalBuilder({
      customId: "ce_description",
      title: "Beschreibung \xE4ndern"
    });
    const description = new TextInputBuilder({
      label: "Gib hier die neue Beschreibung ein",
      minLength: 1,
      customId: "description",
      value: interaction.message.embeds[0].description,
      required: true,
      style: 2
    });
    const row = new ActionRowBuilder().addComponents(description);
    modal.addComponents(row);
    await interaction.showModal(modal);
  }
};
export {
  ce_description_default as default
};
