// contents/dist/ce_footer-GNKNGUYP.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Buttons/CustomEmbed/ce_footer.ts
init_esm_shims();
import { ActionRowBuilder } from "@discordjs/builders";
import { ModalBuilder, TextInputBuilder } from "discord.js";
var ce_footer_default = {
  id: "ce_footer",
  async execute(interaction) {
    if (interaction.user.username != interaction.message.embeds[0].author.name) return interaction.reply({ content: "Du darfst an diesem Embed nichts \xE4ndern", ephemeral: true });
    const modal = new ModalBuilder({
      title: "Footer \xE4ndern",
      customId: "ce_footer"
    });
    const text = new TextInputBuilder({
      customId: "text",
      maxLength: 2048,
      minLength: 1,
      required: true,
      label: "Was soll im Footer stehen?",
      style: 1
    });
    const iconurl = new TextInputBuilder({
      label: "Gib hier die URL zum icon f\xFCr den footer ein",
      customId: "url",
      required: false,
      style: 1
    });
    const row = new ActionRowBuilder().addComponents(text);
    const row2 = new ActionRowBuilder().addComponents(iconurl);
    modal.addComponents(row, row2);
    await interaction.showModal(modal);
  }
};
export {
  ce_footer_default as default
};
