// contents/dist/ce_field-HB6MZ3II.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Buttons/CustomEmbed/ce_field.ts
init_esm_shims();
import { ModalBuilder, TextInputBuilder, ActionRowBuilder } from "discord.js";
var ce_field_default = {
  id: "ce_field",
  async execute(interaction) {
    if (interaction.user.username != interaction.message.embeds[0].author.name) return interaction.reply({ content: "Du darfst an diesem Embed nichts \xE4ndern", ephemeral: true });
    const modal = new ModalBuilder({
      customId: "ce_field",
      title: "Feld \xE4ndern"
    });
    const fieldnr = new TextInputBuilder({
      label: "Welches Feld soll ge\xE4ndert werden",
      minLength: 1,
      maxLength: 2,
      customId: "fieldnr",
      placeholder: "1-25",
      required: true,
      style: 2
    });
    const fieldName = new TextInputBuilder({
      label: "Was soll der Name des Felds sein?",
      //@ts-ignore
      minlength: 1,
      maxLength: 256,
      customId: "fieldname",
      required: true,
      style: 1
    });
    const fieldValue = new TextInputBuilder({
      label: "Was soll in dem Feld stehen?",
      //@ts-ignore
      minlength: 1,
      maxLength: 1024,
      customId: "fieldvalue",
      required: true,
      style: 2
    });
    const row = new ActionRowBuilder().addComponents(fieldnr);
    const row2 = new ActionRowBuilder().addComponents(fieldName);
    const row3 = new ActionRowBuilder().addComponents(fieldValue);
    modal.addComponents(row, row2, row3);
    await interaction.showModal(modal);
  }
};
export {
  ce_field_default as default
};
