// contents/dist/chunk-534IBQPS.js
import {
  casino_default,
  init_casino
} from "./chunk-W44NITJV.js";
import {
  __esm,
  __export,
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Casino/addmoney.ts
var addmoney_exports = {};
__export(addmoney_exports, {
  default: () => addmoney_default
});
import { SlashCommandBuilder, PermissionFlagsBits, EmbedBuilder, Colors } from "discord.js";
var command, addmoney_default;
var init_addmoney = __esm({
  "Commands/Casino/addmoney.ts"() {
    init_esm_shims();
    init_casino();
    command = {
      data: new SlashCommandBuilder().setName("addmoney").setDescription("F\xFCge dem Konto eines Users Geld hinzu").setDefaultMemberPermissions(PermissionFlagsBits.Administrator).addNumberOption((input) => input.setName("amount").setDescription("Wie viel soll hinzugef\xFCgt werden?").setRequired(true)).addUserOption((input) => input.setName("user").setDescription("Der User, der das Geld erh\xE4lt.")),
      async execute(interaction) {
        const user = interaction.options.getUser("user") || interaction.user, amount = interaction.options.getNumber("amount", true);
        const member = interaction.member;
        if (amount < 1) return interaction.reply({ content: "Deine Eingabe ist ung\xFCltig. (Sie darf nicht negativ oder 0 sein)", ephemeral: true });
        const minRole = interaction.guild.roles.cache.get("1170957646942191688");
        if (member.roles.highest.position < minRole.position) return interaction.reply({ embeds: [new EmbedBuilder({ title: "Fehler", description: "Du hast nicht die erforderlichen Berechtigungen f\xFCr diesen Command", color: Colors.DarkRed, timestamp: /* @__PURE__ */ new Date() })] });
        let User2;
        User2 = await casino_default.findOne({ user: user.id });
        if (!User2) {
          User2 = await casino_default.create({
            user: user.id,
            wallet: 0,
            bank: 0,
            inventory: {}
          });
        }
        User2.wallet += amount;
        await User2.save();
        interaction.reply({ content: `Du hast ${user} erfolgreich \u{1F4B0}${amount} gegeben.`, ephemeral: true });
      }
    };
    addmoney_default = command;
  }
});

export {
  addmoney_default,
  addmoney_exports,
  init_addmoney
};
