// contents/dist/chunk-UC232BKO.js
import {
  casino_default,
  init_casino
} from "./chunk-W44NITJV.js";
import {
  __esm,
  __export,
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Casino/withdraw.ts
var withdraw_exports = {};
__export(withdraw_exports, {
  default: () => withdraw_default
});
import { EmbedBuilder, SlashCommandBuilder } from "discord.js";
var command, withdraw_default;
var init_withdraw = __esm({
  "Commands/Casino/withdraw.ts"() {
    init_esm_shims();
    init_casino();
    command = {
      data: new SlashCommandBuilder().setName("withdraw").setDescription("Hebe Geld von deinem Bankkonto ab").addNumberOption((input) => input.setName("amount").setDescription("Wie viel m\xF6chtest du abheben? (-1 f\xFCr alles)").setRequired(true)),
      async execute(interaction) {
        let User = await casino_default.findOne({ user: interaction.user.id });
        if (!User || User.bank <= 0) {
          const embed2 = new EmbedBuilder({
            title: "Abhebung fehlgeschlagen",
            description: "Du kannst nichts von deinem Bankkonto abheben, weil dort kein Geld drauf ist.",
            color: 16716820
          });
          interaction.reply({ embeds: [embed2] });
          return;
        }
        let amount = interaction.options.getNumber("amount", true);
        if (amount > User.bank) {
          const embed2 = new EmbedBuilder({
            title: "Abhebung fehlgeschlagen",
            description: "Du willst mehr Geld abheben, als du hast.",
            color: 16716820
          });
          interaction.reply({ embeds: [embed2] });
          return;
        }
        if (amount == -1) amount = User.bank;
        const embed = new EmbedBuilder({
          title: "Abhebung erfolgreich",
          description: `Du hast erfolgreich \u{1F4B0}${amount} von der Bank abgehoben.`,
          color: 7864064
        });
        User.bank -= amount;
        User.wallet += amount;
        await User.save();
        interaction.reply({ embeds: [embed] });
      }
    };
    withdraw_default = command;
  }
});

export {
  withdraw_default,
  withdraw_exports,
  init_withdraw
};
