// contents/dist/chunk-WADDUS3E.js
import {
  casino_default,
  init_casino
} from "./chunk-W44NITJV.js";
import {
  __esm,
  __export,
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Casino/balance.ts
var balance_exports = {};
__export(balance_exports, {
  default: () => balance_default
});
import { EmbedBuilder, SlashCommandBuilder } from "discord.js";
var command, balance_default;
var init_balance = __esm({
  "Commands/Casino/balance.ts"() {
    init_esm_shims();
    init_casino();
    command = {
      data: new SlashCommandBuilder().setName("balance").setDescription("Lasse dir deinen Kontostand anzeigen").addUserOption((input) => input.setName("user").setDescription("Der Benutzer, dessen Kontostand du sehen m\xF6chtest.")),
      async execute(interaction) {
        const target = interaction.options.getUser("user");
        let User;
        User = await casino_default.findOne({ user: interaction.user.id });
        if (!User) {
          const embed2 = new EmbedBuilder({
            title: "Dein Kontostand",
            fields: [
              { name: "Bargeld", value: "\u{1F4B0}0" },
              { name: "Bankguthaben", value: "\u{1F4B0}0" },
              { name: "Gesamt", value: "\u{1F4B0}0" }
            ],
            color: 16558339
          });
          interaction.reply({ embeds: [embed2] });
          return;
        }
        if (target) {
          const targetMember = interaction.guild.members.cache.get(target.id);
          const User2 = await casino_default.findOne({ user: target.id });
          if (!User2) return interaction.reply({ content: "Dieser User hat momentan noch kein Geld", ephemeral: true });
          const embed2 = new EmbedBuilder({
            title: `Kontostand von ${targetMember.nickname || targetMember.displayName}`,
            fields: [
              { name: "Bargeld", value: `\u{1F4B0}${User2.wallet}` },
              { name: "Bankguthaben", value: `\u{1F4B0}${User2.bank}` },
              { name: "Gesamt", value: `\u{1F4B0}${User2.wallet + User2.bank}` }
            ],
            color: 16558339
          });
          interaction.reply({ embeds: [embed2] });
          return;
        }
        const embed = new EmbedBuilder({
          title: "Dein Kontostand",
          fields: [
            { name: "Bargeld", value: `\u{1F4B0}${User.wallet}` },
            { name: "Bankguthaben", value: `\u{1F4B0}${User.bank}` },
            { name: "Gesamt", value: `\u{1F4B0}${User.wallet + User.bank}` }
          ],
          color: 16558339
        });
        interaction.reply({ embeds: [embed] });
      }
    };
    balance_default = command;
  }
});

export {
  balance_default,
  balance_exports,
  init_balance
};
