// contents/dist/chunk-GP2IZ6EK.js
import {
  casino_default,
  init_casino
} from "./chunk-W44NITJV.js";
import {
  __esm,
  __export,
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Casino/deposit.ts
var deposit_exports = {};
__export(deposit_exports, {
  default: () => deposit_default
});
import { EmbedBuilder, SlashCommandBuilder } from "discord.js";
var command, deposit_default;
var init_deposit = __esm({
  "Commands/Casino/deposit.ts"() {
    init_esm_shims();
    init_casino();
    command = {
      data: new SlashCommandBuilder().setName("deposit").setDescription("Zahle Geld auf dein Bankkonto ein").addNumberOption((input) => input.setName("amount").setDescription("Wie viel m\xF6chtest du einzahlen? (-1 f\xFCr alles)").setRequired(true)),
      async execute(interaction) {
        let User;
        User = await casino_default.findOne({ user: interaction.user.id });
        if (!User || User.wallet <= 0) {
          const embed2 = new EmbedBuilder({
            title: "Einzahlung fehlgeschlagen",
            description: "Du kannst kein Geld auf deine Bank einzahlen, weil du kein Bargeld hast.",
            color: 16716820
          });
          interaction.reply({ embeds: [embed2] });
          return;
        }
        let amount = interaction.options.getNumber("amount", true);
        if (amount > User.wallet) {
          const embed2 = new EmbedBuilder({
            title: "Einzahlung fehlgeschlagen",
            description: "Du willst mehr Geld einzahlen, als du hast.",
            color: 16716820
          });
          interaction.reply({ embeds: [embed2] });
          return;
        } else if (amount == -1) amount = User.wallet;
        User.wallet -= amount;
        User.bank += amount;
        await User.save();
        const embed = new EmbedBuilder({
          title: "Einzahlung erfolgreich",
          description: `Du hast erfolgreich \u{1F4B0}${amount} auf dein Bankkonto eingezahlt.`,
          color: 7864064
        });
        interaction.reply({ embeds: [embed] });
      }
    };
    deposit_default = command;
  }
});

export {
  deposit_default,
  deposit_exports,
  init_deposit
};
