// contents/dist/chunk-C43B5532.js
import {
  casino_default,
  init_casino
} from "./chunk-W44NITJV.js";
import {
  __esm,
  __export,
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Casino/slot.ts
var slot_exports = {};
__export(slot_exports, {
  default: () => slot_default
});
import { EmbedBuilder, SlashCommandBuilder } from "discord.js";
var command, slot_default;
var init_slot = __esm({
  "Commands/Casino/slot.ts"() {
    init_esm_shims();
    init_casino();
    command = {
      data: new SlashCommandBuilder().setName("slot").setDescription("Spiele an einem Spielautomaten, um mit etwas Gl\xFCck etwas Geld zu gewinnen").addNumberOption((option) => option.setName("bet").setDescription("Wie viel soll das Geld betragen?").setRequired(true)),
      async execute(interaction) {
        const bet = interaction.options.getNumber("bet", true);
        let User;
        User = await casino_default.findOne({ user: interaction.user.id });
        if (!User || User.wallet <= 0 || bet > User.wallet) {
          const embed = new EmbedBuilder({
            title: "Spiel fehlgeschlagen",
            description: "Du hast nicht so viel Geld, wie du einsetzen m\xF6chtest!",
            color: 16716820
          });
          interaction.reply({ embeds: [embed] });
          return;
        }
        const symbols = ["\u{1F4B0}", "\u{1F911}", "\u{1F4B8}", "\u{1F4AF}"];
        const result = [];
        for (let i = 0; i < 3; i++) {
          const randomIndex = Math.floor(Math.random() * symbols.length);
          result.push(symbols[randomIndex]);
        }
        const isWin = result[0] == result[1] && result[1] == result[2];
        const winnings = isWin ? bet * 2 : 0;
        User.wallet = User.wallet - bet + winnings;
        await User.save();
        if (winnings == 0) {
          const embed = new EmbedBuilder({
            title: "Leider verloren",
            description: `Du hast \u{1F4B0}${bet} durch folgendes Ergebnis verloren:
${result.join(" | ")}`,
            color: 16716820
          });
          interaction.reply({ embeds: [embed] });
          return;
        } else {
          const embed = new EmbedBuilder({
            title: "Herzlichen Gl\xFCckwunsch",
            description: `Du hast ${winnings} durch folgendes Ergebnis gewonnen:
${result.join(" | ")}`,
            color: 7864064
          });
          interaction.reply({ embeds: [embed] });
          return;
        }
      }
    };
    slot_default = command;
  }
});

export {
  slot_default,
  slot_exports,
  init_slot
};
