// contents/dist/clan-entry-A3SK42RI.js
import "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/Ticket/clan-entry.ts
init_esm_shims();
import { EmbedBuilder, ButtonBuilder, ActionRowBuilder, ChannelType, OverwriteType, ButtonStyle } from "discord.js";
var clan_entry_default = {
  id: "clan-entry",
  async execute(interaction) {
    await interaction.deferReply({ ephemeral: true });
    await interaction.editReply("Ticket wird erstellt...");
    const infos = interaction.fields.getTextInputValue("clan");
    const channel = await interaction.guild.channels.create({
      name: `clan-${interaction.user.username}`,
      parent: "1173314530521129042" /* ticket */,
      type: ChannelType.GuildText,
      permissionOverwrites: [
        { id: interaction.user.id, allow: ["SendMessages", "ViewChannel"], type: OverwriteType.Member },
        { id: interaction.guild.roles.everyone, deny: ["ViewChannel"], type: OverwriteType.Role }
      ]
    });
    const embed = new EmbedBuilder({
      description: `### ${interaction.member} m\xF6chte einem Clan beitreten:
\`\`\`${infos}\`\`\``
    });
    const row = new ActionRowBuilder().addComponents([
      new ButtonBuilder({
        customId: "clanticket-close",
        label: "Ticket schlie\xDFen",
        emoji: "\u{1F512}",
        style: ButtonStyle.Secondary
      })
    ]);
    await channel.send({ content: `${interaction.guild.roles.everyone}`, embeds: [embed], components: [row] });
    await interaction.editReply(`Ticket erstellt: ${channel}`);
  }
};
export {
  clan_entry_default as default
};
