// contents/dist/close-with-reason-JHYKZWPL.js
import "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/Ticket/close-with-reason.ts
init_esm_shims();
import { EmbedBuilder, Colors } from "discord.js";
var close_with_reason_default = {
  id: "close-with-reason",
  async execute(interaction, client) {
    await interaction.deferReply({ ephemeral: true });
    const logChannel = client.channels.cache.get("1173939552239501333" /* ticket_log */);
    const username = interaction.channel.name.split("-")[1];
    const member = interaction.guild.members.cache.find((user) => user.user.username == username);
    const logEmbed = new EmbedBuilder({
      title: "Ticket geschlossen",
      fields: [
        { name: "Ersteller", value: `${member}`, inline: true },
        { name: "Geschlossen von", value: `${interaction.member}`, inline: true },
        { name: "Erstellungsdatum", value: `${interaction.channel?.createdAt?.toLocaleDateString()}`, inline: false },
        { name: "Grund der Schlie\xDFung", value: interaction.fields.getTextInputValue("reason") }
      ],
      color: Colors.Green,
      timestamp: /* @__PURE__ */ new Date()
    });
    const userEmbed = new EmbedBuilder(logEmbed.data).setAuthor({ name: interaction.guild.name, iconURL: interaction.guild.iconURL() || "" });
    await logChannel.send({ embeds: [logEmbed] });
    await member.send({ embeds: [userEmbed] }).catch(async (err) => {
      await interaction.editReply("Die DM Nachricht konnte nicht zugestellt werden!");
    }).then(async () => {
      await interaction.editReply("Der Ersteller des Tickets wurde \xFCber die Schlie\xDFung informiert!");
    }).finally(async () => {
      await interaction.channel.send({ content: "Dieses Ticket wird in 5 Sekunden gel\xF6scht!" });
    });
    setTimeout(async () => {
      await interaction.channel.delete();
    }, 5e3);
  }
};
export {
  close_with_reason_default as default
};
