// contents/dist/close-UUEOLAXG.js
import "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Buttons/ticket/close.ts
init_esm_shims();
import { EmbedBuilder, ActionRowBuilder, ButtonBuilder, Colors } from "discord.js";
var close_default = {
  id: "close",
  async execute(interaction) {
    if (!interaction.member.roles.cache.has("1156298949301379212" /* staff */)) return interaction.reply({ content: "Nur Teammitglieder d\xFCrfen Tickets schlie\xDFen", ephemeral: true });
    const embed = new EmbedBuilder({
      title: "Schlie\xDFung best\xE4tigen",
      description: "Bitte best\xE4tige mit dem Button unter dieser Nachricht, dass du das Ticket schlie\xDFen m\xF6chtest",
      color: Colors.Blue
    });
    const row = new ActionRowBuilder().addComponents([
      new ButtonBuilder({
        label: "Best\xE4tigen",
        style: 1,
        customId: "close-confirm"
      })
    ]);
    await interaction.reply({ embeds: [embed], components: [row] });
  }
};
export {
  close_default as default
};
