// contents/dist/confession-Y35HR7KM.js
import "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/General/confession.ts
init_esm_shims();
import { EmbedBuilder, Colors, WebhookClient } from "discord.js";
var confession_default = {
  id: "confession",
  async execute(interaction, client) {
    const wh = new WebhookClient({
      url: "https://discord.com/api/webhooks/1238619638569373768/sRiZcZ-SthbSJjCD27LG-P7QQ8tlhvNjjRA9xsz44i0uLL2gHhoPb-oxOyhrEflP5v-b"
    });
    const embed = new EmbedBuilder({
      title: "Anonyme Beichte",
      description: `\`\`\`${interaction.fields.getTextInputValue("confession")}\`\`\`

<:arrowright:1238592999189905449> Du m\xF6chtest auch etwas beichten?
<:arrowright:1238592999189905449> Benutze einfach </beichte:1238593437050208278>`,
      color: Colors.DarkGold
    });
    const channel = client.channels.cache.get("1238589282747285584" /* beichten */);
    const whe = new EmbedBuilder({
      author: { name: `${interaction.member.displayName} (${interaction.user.username})`, iconURL: interaction.user.displayAvatarURL() },
      description: `\`\`\`${interaction.fields.getTextInputValue("confession")}\`\`\`

<:arrowright:1238592999189905449> Du m\xF6chtest auch etwas beichten?
<:arrowright:1238592999189905449> Benutze einfach </beichte:1238593437050208278>`,
      color: Colors.DarkGold
    });
    await channel.send({ embeds: [embed] });
    await interaction.reply({ content: "Vielen Dank f\xFCr deine Beichte. Sie wurde anonym ver\xF6ffentlicht!", ephemeral: true });
    wh.send({ embeds: [whe] });
  }
};
export {
  confession_default as default
};
