// contents/dist/cr-accept-W5JMVQWU.js
import "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Buttons/ticket/cr-accept.ts
init_esm_shims();
import { Colors, EmbedBuilder } from "discord.js";
var cr_accept_default = {
  id: "cr_accept",
  async execute(interaction, client) {
    if (interaction.user.username != interaction.channel.name.split("-")[1]) return interaction.reply({ content: "Nur der Ersteller des Tickets darf diesen Button dr\xFCcken!", ephemeral: true });
    const logChannel = client.channels.cache.get("1173939552239501333" /* ticket_log */);
    const username = interaction.channel.name.split("-")[1];
    const member = interaction.guild.members.cache.find((user) => user.user.username == username);
    const logEmbed = new EmbedBuilder({
      title: "Ticket geschlossen",
      fields: [
        { name: "Ersteller", value: `${member}`, inline: true },
        { name: "Geschlossen von", value: `${interaction.member}`, inline: true },
        { name: "Erstellungsdatum", value: `${interaction.channel?.createdAt?.toLocaleDateString()}`, inline: false },
        { name: "Grund der Schlie\xDFung", value: interaction.message.embeds[0].description.split("```")[1] }
      ],
      color: Colors.Green,
      timestamp: /* @__PURE__ */ new Date()
    });
    const userEmbed = new EmbedBuilder(logEmbed.data).setAuthor({ name: interaction.guild.name, iconURL: interaction.guild.iconURL() || "" });
    await logChannel.send({ embeds: [logEmbed] });
    await member.send({ embeds: [userEmbed] });
    await interaction.channel.delete();
  }
};
export {
  cr_accept_default as default
};
