// contents/dist/customembed-2LE5VIPT.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Admin/customembed.ts
init_esm_shims();
import { SlashCommandBuilder, EmbedBuilder, ButtonBuilder, ActionRowBuilder, PermissionFlagsBits } from "discord.js";
var command = {
  data: new SlashCommandBuilder().setName("customembed").setDescription("Custom Embed System").setDefaultMemberPermissions(PermissionFlagsBits.Administrator).addSubcommand(
    (command2) => command2.setName("create").setDescription("Erstelle ein Custom Embed")
  ),
  async execute(interaction) {
    const embed = new EmbedBuilder({
      author: { name: interaction.user.username, iconURL: interaction.user.displayAvatarURL() },
      title: "Custom Embed",
      description: "Nutze die Buttons, um deinen Embed zu gestalten! :)"
    });
    const row = new ActionRowBuilder().addComponents([
      new ButtonBuilder({
        customId: "ce_title",
        label: "Titel \xE4ndern",
        style: 1
      }),
      new ButtonBuilder({
        customId: "ce_description",
        label: "Beschreibung \xE4ndern",
        style: 1
      }),
      new ButtonBuilder({
        customId: "ce_field",
        label: "Feld \xE4ndern",
        style: 1
      }),
      new ButtonBuilder({
        customId: "ce_thumbnail",
        label: "Thumbnail \xE4ndern",
        style: 1
      }),
      new ButtonBuilder({
        customId: "ce_image",
        label: "Bild \xE4ndern",
        style: 1
      })
    ]);
    const row2 = new ActionRowBuilder().addComponents([
      new ButtonBuilder({
        customId: "ce_color",
        label: "Farbe \xE4ndern",
        style: 1
      }),
      new ButtonBuilder({
        customId: "ce_footer",
        label: "Footer \xE4ndern",
        style: 1
      }),
      new ButtonBuilder({
        customId: "ce_send",
        label: "In einen Channel Senden",
        style: 3
      }),
      new ButtonBuilder({
        customId: "ce_delete",
        label: "Embed l\xF6schen",
        style: 4
      })
    ]);
    interaction.reply({ embeds: [embed], components: [row, row2] });
  }
};
var customembed_default = command;
export {
  customembed_default as default
};
