// contents/dist/decline-PXS4UALB.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Buttons/Abmeldung/decline.ts
init_esm_shims();
import { ModalBuilder, TextInputBuilder, TextInputStyle, ActionRowBuilder } from "discord.js";
var allowed = [
  "773072144304963624",
  "1196808890227118201",
  "1035799257409667102"
];
var decline_default = {
  id: "decline",
  async execute(interaction) {
    if (allowed.some((id) => interaction.user.id == id)) {
      const modal = new ModalBuilder({
        customId: "decline",
        title: "Abmeldung ablehnen"
      });
      const reason = new TextInputBuilder({
        customId: "reason",
        label: "Grund",
        required: false,
        maxLength: 1024,
        style: TextInputStyle.Paragraph
      });
      const row = new ActionRowBuilder().addComponents(reason);
      modal.addComponents(row);
      await interaction.showModal(modal);
    } else return interaction.reply({ content: "Du darfst keine Abmeldungen ablehnen!", ephemeral: true });
  }
};
export {
  decline_default as default
};
