// contents/dist/giveaway-I6EOA4F6.js
import {
  countdown
} from "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Admin/giveaway.ts
init_esm_shims();
import { EmbedBuilder, ButtonBuilder, ActionRowBuilder, PermissionFlagsBits, SlashCommandBuilder, Colors } from "discord.js";
import { readFileSync, writeFileSync } from "fs";
var command = {
  data: new SlashCommandBuilder().setName("giveaway").setDescription("Starte ein Giveaway").setDefaultMemberPermissions(PermissionFlagsBits.Administrator).addStringOption((input) => input.setName("prize").setDescription("Was gibt es zu gewinnen?").setRequired(true)).addNumberOption((input) => input.setName("duration").setDescription("Die Laufzeit in Stunden").setRequired(true)),
  async execute(interaction, client) {
    const giveaways = JSON.parse(readFileSync("./JSON/giveaways.json", "utf8"));
    const prize = interaction.options.getString("prize", true);
    const duration = interaction.options.getNumber("duration", true) * 36e5;
    const finalDuration = Date.now() + duration;
    const embed = new EmbedBuilder({
      title: "Neues Giveaway",
      description: `Es gibt **__${prize}__** zu gewinnen!`,
      fields: [
        { name: "Teilnehmer", value: "0" },
        { name: "Endet", value: countdown(finalDuration) }
      ],
      color: Colors.Gold
    });
    const row = new ActionRowBuilder({
      components: [
        new ButtonBuilder({
          label: "Teilnehmen",
          emoji: "\u{1F389}",
          customId: "giveaway_entry",
          style: 2
        })
      ]
    });
    const channel = client.guild.channels.cache.get("1178073046444163102" /* test */);
    const message = await channel.send({ embeds: [embed], content: "<@&1148638318263799869>" /* giveaway */, components: [row] });
    const giveaway = {
      prize,
      endTime: finalDuration,
      messageId: message.id,
      participants: []
    };
    giveaways.push(giveaway);
    writeFileSync("./JSON/giveaways.json", JSON.stringify(giveaways, null, 2));
    await interaction.reply({ content: " Das Giveaway wurde erfolgreich erstellt!", ephemeral: true });
  }
};
var giveaway_default = command;
export {
  giveaway_default as default
};
