// contents/dist/help-O64ZV5M4.js
import {
  init_slot,
  slot_exports
} from "./chunk-C43B5532.js";
import {
  init_transaction,
  transaction_exports
} from "./chunk-YY34732D.js";
import {
  init_withdraw,
  withdraw_exports
} from "./chunk-UC232BKO.js";
import {
  init_work,
  work_exports
} from "./chunk-HOOKTH55.js";
import {
  coinflip_exports,
  init_coinflip
} from "./chunk-SD5BOIDZ.js";
import {
  crime_exports,
  init_crime
} from "./chunk-I642TC43.js";
import {
  daily_exports,
  init_daily
} from "./chunk-DLE3QELZ.js";
import {
  deposit_exports,
  init_deposit
} from "./chunk-GP2IZ6EK.js";
import {
  init_pay,
  pay_exports
} from "./chunk-6HESBH6M.js";
import {
  init_removemoney,
  removemoney_exports
} from "./chunk-LDXFX5NF.js";
import {
  init_reset,
  reset_exports
} from "./chunk-YJ46JNSZ.js";
import {
  init_rob,
  rob_exports
} from "./chunk-43JDU7LM.js";
import "./chunk-NJJFVLCF.js";
import {
  addmoney_exports,
  init_addmoney
} from "./chunk-534IBQPS.js";
import {
  balance_exports,
  init_balance
} from "./chunk-WADDUS3E.js";
import "./chunk-W44NITJV.js";
import {
  __glob,
  __toCommonJS,
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/General/help.ts
init_esm_shims();
import { EmbedBuilder, SlashCommandBuilder } from "discord.js";
import fs from "fs";

// require("../../Commands/Casino/**/*") in Commands/General/help.ts
var globRequire_Commands_Casino = __glob({
  "../../Commands/Casino/addmoney.ts": () => (init_addmoney(), __toCommonJS(addmoney_exports)),
  "../../Commands/Casino/balance.ts": () => (init_balance(), __toCommonJS(balance_exports)),
  "../../Commands/Casino/coinflip.ts": () => (init_coinflip(), __toCommonJS(coinflip_exports)),
  "../../Commands/Casino/crime.ts": () => (init_crime(), __toCommonJS(crime_exports)),
  "../../Commands/Casino/daily.ts": () => (init_daily(), __toCommonJS(daily_exports)),
  "../../Commands/Casino/deposit.ts": () => (init_deposit(), __toCommonJS(deposit_exports)),
  "../../Commands/Casino/pay.ts": () => (init_pay(), __toCommonJS(pay_exports)),
  "../../Commands/Casino/removemoney.ts": () => (init_removemoney(), __toCommonJS(removemoney_exports)),
  "../../Commands/Casino/reset.ts": () => (init_reset(), __toCommonJS(reset_exports)),
  "../../Commands/Casino/rob.ts": () => (init_rob(), __toCommonJS(rob_exports)),
  "../../Commands/Casino/slot.ts": () => (init_slot(), __toCommonJS(slot_exports)),
  "../../Commands/Casino/transaction.ts": () => (init_transaction(), __toCommonJS(transaction_exports)),
  "../../Commands/Casino/withdraw.ts": () => (init_withdraw(), __toCommonJS(withdraw_exports)),
  "../../Commands/Casino/work.ts": () => (init_work(), __toCommonJS(work_exports))
});

// Commands/General/help.ts
var command = {
  data: new SlashCommandBuilder().setName("help").setDescription("F\xFChre diesen Befehl aus, wenn du Hilfe brauchst").addSubcommand(
    (input) => input.setName("casino").setDescription("Lasse dir alle Befehle zum Casino System anzeigen")
  ),
  async execute(interaction, client) {
    const subcommand = interaction.options.getSubcommand();
    switch (subcommand) {
      case "casino": {
        const fieldArray = [];
        const dir = fs.readdirSync("./Commands/Casino");
        dir.forEach((file) => {
          const module = globRequire_Commands_Casino(`../../Commands/Casino/${file}`);
          if (!module.cmdid) return;
          const name = module.name;
          const value = module.description;
          fieldArray.push({ name: `</${name}:${module.cmdid}>`, value });
        });
        const embed = new EmbedBuilder({
          title: "**Casino Hilfe**",
          description: "Hier findest du alle Befehle des Casino Systems",
          fields: fieldArray,
          color: 16761856
        });
        interaction.reply({ embeds: [embed] });
      }
    }
  }
};
var help_default = command;
export {
  help_default as default
};
