// contents/dist/index.js
import {
  __glob,
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// src/index.ts
init_esm_shims();

// src/contents.ts
init_esm_shims();
import { Client, Collection } from "discord.js";
import { createAudioPlayer } from "@discordjs/voice";
import { readdirSync } from "fs";
import chalk from "chalk";

// import("../SelectMenus/**/*/**/*") in src/contents.ts
var globImport_SelectMenus = __glob({
  "../SelectMenus/Ticket/ticket.ts": () => import("./ticket-AURYP4BD.js")
});

// import("../Commands/**/*/**/*") in src/contents.ts
var globImport_Commands = __glob({
  "../Commands/Admin/backup.ts": () => import("./backup-JW7DKXFM.js"),
  "../Commands/Admin/changestatus.ts": () => import("./changestatus-HQ4D52AB.js"),
  "../Commands/Admin/closerequest.ts": () => import("./closerequest-2ZS67GHO.js"),
  "../Commands/Admin/customembed.ts": () => import("./customembed-2LE5VIPT.js"),
  "../Commands/Admin/emoji.ts": () => import("./emoji-5SDQWV6O.js"),
  "../Commands/Admin/getOwner.ts": () => import("./getOwner-62DCHIUA.js"),
  "../Commands/Admin/giveaway.ts": () => import("./giveaway-I6EOA4F6.js"),
  "../Commands/Admin/logout.ts": () => import("./logout-74CE3AOQ.js"),
  "../Commands/Admin/perms.ts": () => import("./perms-REXAQX2U.js"),
  "../Commands/Admin/poll.ts": () => import("./poll-IOLFWIQX.js"),
  "../Commands/Admin/resetmessages.ts": () => import("./resetmessages-6Y2NJZDV.js"),
  "../Commands/Admin/resetstatus.ts": () => import("./resetstatus-UQNPWFKO.js"),
  "../Commands/Admin/roles.ts": () => import("./roles-ELYRXFZM.js"),
  "../Commands/Admin/rules.ts": () => import("./rules-NQMQZ7JY.js"),
  "../Commands/Admin/selfrole.ts": () => import("./selfrole-UKVOFCYG.js"),
  "../Commands/Admin/staff-changes.ts": () => import("./staff-changes-OQVGX5OQ.js"),
  "../Commands/Admin/task.ts": () => import("./task-T2GZI5SL.js"),
  "../Commands/Admin/test.ts": () => import("./test-ML74SO3G.js"),
  "../Commands/Admin/ticket.ts": () => import("./ticket-5B2UJ7S5.js"),
  "../Commands/Admin/updatestafflist.ts": () => import("./updatestafflist-IOYWPKN4.js"),
  "../Commands/Casino/addmoney.ts": () => import("./addmoney-AU2TKGLZ.js"),
  "../Commands/Casino/balance.ts": () => import("./balance-2FKGSWJD.js"),
  "../Commands/Casino/coinflip.ts": () => import("./coinflip-LL67DRHA.js"),
  "../Commands/Casino/crime.ts": () => import("./crime-NXYNYNAR.js"),
  "../Commands/Casino/daily.ts": () => import("./daily-2I4OPLL6.js"),
  "../Commands/Casino/deposit.ts": () => import("./deposit-HPAOIY3R.js"),
  "../Commands/Casino/pay.ts": () => import("./pay-J2MEHQ4R.js"),
  "../Commands/Casino/removemoney.ts": () => import("./removemoney-7SPCIEYU.js"),
  "../Commands/Casino/reset.ts": () => import("./reset-NSYWAQH7.js"),
  "../Commands/Casino/rob.ts": () => import("./rob-PQIGOEKA.js"),
  "../Commands/Casino/slot.ts": () => import("./slot-NDEA2LCD.js"),
  "../Commands/Casino/transaction.ts": () => import("./transaction-OJDJ3QRD.js"),
  "../Commands/Casino/withdraw.ts": () => import("./withdraw-TMNBTC3P.js"),
  "../Commands/Casino/work.ts": () => import("./work-3GERSBUX.js"),
  "../Commands/General/bumps.ts": () => import("./bumps-BJSSK46X.js"),
  "../Commands/General/commands.ts": () => import("./commands-MGT62BE3.js"),
  "../Commands/General/confession.ts": () => import("./confession-CP5K5O6Q.js"),
  "../Commands/General/help.ts": () => import("./help-O64ZV5M4.js"),
  "../Commands/General/info.ts": () => import("./info-CYPXQCYK.js"),
  "../Commands/General/report.ts": () => import("./report-KBQZ47KT.js"),
  "../Commands/General/suggestion.ts": () => import("./suggestion-IZQVFWKW.js"),
  "../Commands/Moderation/addrole.ts": () => import("./addrole-6GIIDXBJ.js"),
  "../Commands/Moderation/ban.ts": () => import("./ban-U2VQYGHE.js"),
  "../Commands/Moderation/clearwarnings.ts": () => import("./clearwarnings-Z5Z3LK3Z.js"),
  "../Commands/Moderation/fullclear.ts": () => import("./fullclear-H4PTE6CP.js"),
  "../Commands/Moderation/kick.ts": () => import("./kick-DG5FGMLS.js"),
  "../Commands/Moderation/unwarn.ts": () => import("./unwarn-XJNZTJL6.js"),
  "../Commands/Moderation/warn.ts": () => import("./warn-GH27KCTN.js"),
  "../Commands/Moderation/warns.ts": () => import("./warns-SPTTTM7W.js"),
  "../Commands/Music/music.ts": () => import("./music-T53YPHGE.js"),
  "../Commands/Staff/abmeldung.ts": () => import("./abmeldung-N3IFGXWD.js"),
  "../Commands/Staff/messages.ts": () => import("./messages-2M2ILVTV.js")
});

// import("../Buttons/**/*/**/*") in src/contents.ts
var globImport_Buttons = __glob({
  "../Buttons/Abmeldung/accept.ts": () => import("./accept-GN2Q5XEM.js"),
  "../Buttons/Abmeldung/decline.ts": () => import("./decline-PXS4UALB.js"),
  "../Buttons/Admin/claimtask.ts": () => import("./claimtask-SYJMRUSL.js"),
  "../Buttons/Admin/finishtask.ts": () => import("./finishtask-3M63GVP4.js"),
  "../Buttons/Admin/updatetask.ts": () => import("./updatetask-6YSZ3SWT.js"),
  "../Buttons/CustomEmbed/ce_color.ts": () => import("./ce_color-ANOU2UFD.js"),
  "../Buttons/CustomEmbed/ce_delete.ts": () => import("./ce_delete-M7SSOF2K.js"),
  "../Buttons/CustomEmbed/ce_description.ts": () => import("./ce_description-SCOMUBEG.js"),
  "../Buttons/CustomEmbed/ce_field.ts": () => import("./ce_field-HB6MZ3II.js"),
  "../Buttons/CustomEmbed/ce_footer.ts": () => import("./ce_footer-GNKNGUYP.js"),
  "../Buttons/CustomEmbed/ce_image.ts": () => import("./ce_image-EFNCVD6Y.js"),
  "../Buttons/CustomEmbed/ce_send.ts": () => import("./ce_send-ZNMSINUL.js"),
  "../Buttons/CustomEmbed/ce_thumbnail.ts": () => import("./ce_thumbnail-3FURGKIE.js"),
  "../Buttons/CustomEmbed/ce_title.ts": () => import("./ce_title-YGA2LXSG.js"),
  "../Buttons/General/giveaway_entry.ts": () => import("./giveaway_entry-R7TXPGFG.js"),
  "../Buttons/General/greet.ts": () => import("./greet-TKZYBPRS.js"),
  "../Buttons/General/verify.ts": () => import("./verify-7PP2TGM3.js"),
  "../Buttons/Security/unban-request.ts": () => import("./unban-request-CRH6ZD3A.js"),
  "../Buttons/Selfrole/13+.ts": () => import("./13+-ZKCUACTD.js"),
  "../Buttons/Selfrole/15+.ts": () => import("./15+-LF2BNP2O.js"),
  "../Buttons/Selfrole/18+.ts": () => import("./18+-3JHDGLTX.js"),
  "../Buttons/Selfrole/20+.ts": () => import("./20+-5OSXZWHP.js"),
  "../Buttons/Selfrole/allgemeines.ts": () => import("./allgemeines-F2SKZB4B.js"),
  "../Buttons/Selfrole/androis.ts": () => import("./androis-L43KRSZP.js"),
  "../Buttons/Selfrole/botstatus.ts": () => import("./botstatus-NSHEARJJ.js"),
  "../Buttons/Selfrole/brawlstars.ts": () => import("./brawlstars-IKMDZMNB.js"),
  "../Buttons/Selfrole/bumping.ts": () => import("./bumping-S5J3JFQZ.js"),
  "../Buttons/Selfrole/coc.ts": () => import("./coc-4TF6DZKT.js"),
  "../Buttons/Selfrole/cod.ts": () => import("./cod-KLJGDYB7.js"),
  "../Buttons/Selfrole/events.ts": () => import("./events-YTMUYU7P.js"),
  "../Buttons/Selfrole/female.ts": () => import("./female-XELDSGBJ.js"),
  "../Buttons/Selfrole/fortnite.ts": () => import("./fortnite-XEQXB5TY.js"),
  "../Buttons/Selfrole/frankreich.ts": () => import("./frankreich-C6WJ4YA7.js"),
  "../Buttons/Selfrole/germany.ts": () => import("./germany-4DMIJXA7.js"),
  "../Buttons/Selfrole/giveaways.ts": () => import("./giveaways-3IX2INCN.js"),
  "../Buttons/Selfrole/gta.ts": () => import("./gta-HSP24O7G.js"),
  "../Buttons/Selfrole/ios.ts": () => import("./ios-KNLJSBYK.js"),
  "../Buttons/Selfrole/italien.ts": () => import("./italien-PKDYPFPG.js"),
  "../Buttons/Selfrole/konsole.ts": () => import("./konsole-WDOY5MB7.js"),
  "../Buttons/Selfrole/male.ts": () => import("./male-LVXBB4YL.js"),
  "../Buttons/Selfrole/mcbedrock.ts": () => import("./mcbedrock-6QUF5TRJ.js"),
  "../Buttons/Selfrole/mcjava.ts": () => import("./mcjava-B7QNB2EA.js"),
  "../Buttons/Selfrole/news.ts": () => import("./news-VLZLMH4O.js"),
  "../Buttons/Selfrole/niederlande.ts": () => import("./niederlande-WKL4AFYF.js"),
  "../Buttons/Selfrole/pc.ts": () => import("./pc-2EQPAT3F.js"),
  "../Buttons/Selfrole/polls.ts": () => import("./polls-ITZV7CVL.js"),
  "../Buttons/Selfrole/rocketleague.ts": () => import("./rocketleague-7SHC75UA.js"),
  "../Buttons/Selfrole/schweiz.ts": () => import("./schweiz-UDP6C7OX.js"),
  "../Buttons/Selfrole/staffchanges.ts": () => import("./staffchanges-QGXVS3QD.js"),
  "../Buttons/Selfrole/valorant.ts": () => import("./valorant-IS2TPCJP.js"),
  "../Buttons/Selfrole/wop.ts": () => import("./wop-KNKLULCB.js"),
  "../Buttons/Selfrole/\xF6sterreich.ts": () => import("./österreich-WKYPBXJT.js"),
  "../Buttons/ticket/claim.ts": () => import("./claim-BAKS5OPP.js"),
  "../Buttons/ticket/clan-close.ts": () => import("./clan-close-N4EIRXBR.js"),
  "../Buttons/ticket/clan.ts": () => import("./clan-YRA3NJLI.js"),
  "../Buttons/ticket/close-confirm.ts": () => import("./close-confirm-6C7NJLAV.js"),
  "../Buttons/ticket/close-decline.ts": () => import("./close-decline-CDAQ3E32.js"),
  "../Buttons/ticket/close-with-reason.ts": () => import("./close-with-reason-A677OJXW.js"),
  "../Buttons/ticket/close.ts": () => import("./close-UUEOLAXG.js"),
  "../Buttons/ticket/cr-accept.ts": () => import("./cr-accept-W5JMVQWU.js"),
  "../Buttons/ticket/cr-decline.ts": () => import("./cr-decline-UT4PYTBW.js")
});

// import("../Modals/**/*/**/*") in src/contents.ts
var globImport_Modals = __glob({
  "../Modals/Abmeldung/accept.ts": () => import("./accept-TYJJUFZP.js"),
  "../Modals/Abmeldung/decline.ts": () => import("./decline-75X6QHHN.js"),
  "../Modals/Admin/task.ts": () => import("./task-MZN5OEZQ.js"),
  "../Modals/Admin/taskupdate.ts": () => import("./taskupdate-DZQTZFZ6.js"),
  "../Modals/CustomEmbed/ce_color.ts": () => import("./ce_color-ZBHFR6BN.js"),
  "../Modals/CustomEmbed/ce_description.ts": () => import("./ce_description-OEYRHYBH.js"),
  "../Modals/CustomEmbed/ce_field.ts": () => import("./ce_field-WRQD5CVT.js"),
  "../Modals/CustomEmbed/ce_footer.ts": () => import("./ce_footer-MZCJFZNZ.js"),
  "../Modals/CustomEmbed/ce_image.ts": () => import("./ce_image-UGZY57LQ.js"),
  "../Modals/CustomEmbed/ce_send.ts": () => import("./ce_send-VCADK47A.js"),
  "../Modals/CustomEmbed/ce_thumbnail.ts": () => import("./ce_thumbnail-V5FWRMR3.js"),
  "../Modals/CustomEmbed/ce_title.ts": () => import("./ce_title-YTGTVYF3.js"),
  "../Modals/General/confession.ts": () => import("./confession-Y35HR7KM.js"),
  "../Modals/General/suggestion.ts": () => import("./suggestion-5FTHACHP.js"),
  "../Modals/Security/unban-request.ts": () => import("./unban-request-F33J5DQG.js"),
  "../Modals/Staff/abmeldung.ts": () => import("./abmeldung-BMCGIIZL.js"),
  "../Modals/Ticket/apply.ts": () => import("./apply-IPSJK42P.js"),
  "../Modals/Ticket/clan-entry.ts": () => import("./clan-entry-A3SK42RI.js"),
  "../Modals/Ticket/close-with-reason.ts": () => import("./close-with-reason-JHYKZWPL.js"),
  "../Modals/Ticket/partnership.ts": () => import("./partnership-I4ZK5PXB.js"),
  "../Modals/Ticket/report.ts": () => import("./report-5V5JHHIR.js"),
  "../Modals/Ticket/support.ts": () => import("./support-MWYEEGSK.js")
});

// import("../ContextMenus/**/*/**/*") in src/contents.ts
var globImport_ContextMenus = __glob({
  "../ContextMenus/Messages/getMessages.ts": () => import("./getMessages-JMHTRNBN.js"),
  "../ContextMenus/Warns/getWarns.ts": () => import("./getWarns-CHFFGBHR.js")
});

// import("../Events/**/*/**/*") in src/contents.ts
var globImport_Events = __glob({
  "../Events/Admin/devTaskDel.ts": () => import("./devTaskDel-HXSK5B6S.js"),
  "../Events/Client/ready.ts": () => import("./ready-XK4YJA7V.js"),
  "../Events/Database/connect.ts": () => import("./connect-DU4VZQDO.js"),
  "../Events/General/bump-reminder.ts": () => import("./bump-reminder-P6XVTNV5.js"),
  "../Events/General/crosspost.ts": () => import("./crosspost-4ZDMT35M.js"),
  "../Events/General/dmlog.ts": () => import("./dmlog-SQ6WDYQA.js"),
  "../Events/General/join.ts": () => import("./join-HE2LRAYU.js"),
  "../Events/Observer/dc-stats.ts": () => import("./dc-stats-RNWAN4DQ.js"),
  "../Events/Observer/giveaways.ts": () => import("./giveaways-VFUROMCY.js"),
  "../Events/Observer/staff-changes.ts": () => import("./staff-changes-PV6SOPHY.js"),
  "../Events/Security/anti-selfbot.ts": () => import("./anti-selfbot-6JDOXWKL.js"),
  "../Events/Security/autokick.ts": () => import("./autokick-CUGTA5Y3.js"),
  "../Events/messageCounter/addUser.ts": () => import("./addUser-6JGYF6DQ.js"),
  "../Events/messageCounter/countMessages.ts": () => import("./countMessages-ISJ5UKMQ.js"),
  "../Events/messageCounter/messageCounterreset.ts": () => import("./messageCounterreset-USMAORJ4.js")
});

// src/contents.ts
var ConsoleInfo = class {
  show(message) {
    console.log(chalk.greenBright(`[${(/* @__PURE__ */ new Date()).toLocaleTimeString("de")} INFO]`), chalk.whiteBright(message));
  }
};
var ConsoleWarning = class {
  show(message) {
    console.log(chalk.yellowBright(`[${(/* @__PURE__ */ new Date()).toLocaleTimeString("de")} WARN] ${message}`));
  }
};
var MyClient = class extends Client {
  commands;
  apps;
  contextMenus;
  modals;
  selectMenus;
  buttons;
  guild;
  queue;
  player;
  constructor(options) {
    super(options);
    this.commands = new Collection();
    this.apps = [];
    this.contextMenus = new Collection();
    this.modals = new Collection();
    this.selectMenus = new Collection();
    this.buttons = new Collection();
    this.queue = [];
  }
  setGuild(guild) {
    this.guild = guild;
  }
  enableAudioPlayer() {
    this.player = createAudioPlayer();
    console.log("AudioPlayer aktiviert");
  }
  async loadAll() {
    const selectMenuSubdirs = readdirSync("./SelectMenus");
    for (const dir of selectMenuSubdirs) {
      const files = readdirSync(`./SelectMenus/${dir}`);
      for (const file of files) {
        const module = await globImport_SelectMenus(`../SelectMenus/${dir}/${file}`);
        const menu = module.default;
        if (!menu || !menu.id) {
          new ConsoleWarning().show(`Fehler bei Select Menu in SelectMenus/${dir}/${file}`);
          continue;
        }
        this.selectMenus.set(menu.id, menu);
        new ConsoleInfo().show(`Select Menu "${menu.id}" geladen`);
      }
    }
    const commandSubdirs = readdirSync("./Commands");
    for (const dir of commandSubdirs) {
      const files = readdirSync(`./Commands/${dir}`);
      for (const file of files) {
        const module = await globImport_Commands(`../Commands/${dir}/${file}`);
        const command = module.default;
        if (!command || !command.data || !command.data.name || !command.data.description) {
          new ConsoleInfo().show(`Befehl in Commands/${dir}/${file} ist ung\xFCltig`);
          continue;
        }
        this.commands.set(command.data.name, command);
        this.apps.push(command);
        new ConsoleInfo().show(`Befehl "/${command.data.name}" geladen`);
      }
    }
    const buttonSubdirs = readdirSync("./Buttons");
    for (const dir of buttonSubdirs) {
      const files = readdirSync(`./Buttons/${dir}`);
      for (const file of files) {
        const module = await globImport_Buttons(`../Buttons/${dir}/${file}`);
        const button = module.default;
        if (!button || !button.id) {
          new ConsoleWarning().show(`Fehler bei Button in Buttons/${dir}/${file}`);
          continue;
        }
        this.buttons.set(button.id, button);
        new ConsoleInfo().show(`Button "${button.id}" geladen`);
      }
    }
    const modalSubdirs = readdirSync("./Modals");
    for (const dir of modalSubdirs) {
      const files = readdirSync(`./Modals/${dir}`);
      for (const file of files) {
        const module = await globImport_Modals(`../Modals/${dir}/${file}`);
        const modal = module.default;
        if (!modal || !modal.id) {
          new ConsoleWarning().show(`Fehler bei Modal in Modals/${dir}/${file}`);
          continue;
        }
        this.modals.set(modal.id, modal);
        new ConsoleInfo().show(`Modal "${modal.id}" geladen`);
      }
    }
    const contextMenuSubdirs = readdirSync("./ContextMenus");
    for (const dir of contextMenuSubdirs) {
      const files = readdirSync(`./ContextMenus/${dir}`);
      for (const file of files) {
        const module = await globImport_ContextMenus(`../ContextMenus/${dir}/${file}`);
        const menu = module.default;
        if (!menu || !menu.data.name || !menu.data.type) {
          new ConsoleWarning().show(`Fehler bei Context Menu in ContextMenus/${dir}/${file}`);
          continue;
        }
        this.contextMenus.set(menu.data.name, menu);
        this.apps.push(menu);
        new ConsoleInfo().show(`ContextMenu "${menu.data.name}" geladen`);
      }
    }
    const eventSubdirs = readdirSync("./Events");
    for (const dir of eventSubdirs) {
      const files = readdirSync(`./Events/${dir}`);
      for (const file of files) {
        const module = await globImport_Events(`../Events/${dir}/${file}`);
        const event = module.default;
        if (!event || !event.name) {
          new ConsoleWarning().show(`Fehler bei Event in Events/${dir}/${file}`);
          continue;
        }
        if (event.once) {
          this.once(event.name, (...args) => event.execute(...args, this));
        } else {
          this.on(event.name, (...args) => event.execute(...args, this));
        }
        new ConsoleInfo().show(`${event.name}-Event "${file.split(".")[0]}" geladen`);
      }
    }
  }
};
export {
  ConsoleInfo,
  ConsoleWarning,
  MyClient
};
