// contents/dist/partnership-I4ZK5PXB.js
import {
  ticketButtons
} from "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/Ticket/partnership.ts
init_esm_shims();
import { EmbedBuilder, ChannelType } from "discord.js";
var partnership_default = {
  id: "partnerschaft",
  async execute(interaction, client) {
    const channel = await interaction.guild.channels.create({
      name: `ticket-${interaction.user.username}`,
      type: ChannelType.GuildText,
      parent: "1173314530521129042" /* ticket */,
      permissionOverwrites: [
        { id: interaction.user, allow: ["SendMessages", "ViewChannel"] },
        { id: interaction.guild.roles.everyone, deny: ["ViewChannel"] },
        { id: "1156298949301379212" /* staff */, allow: ["SendMessages", "ViewChannel"] }
      ]
    });
    const memberCount = interaction.fields.getTextInputValue("membercount");
    const description = interaction.fields.getTextInputValue("description");
    const embed = new EmbedBuilder({
      description: `Dies ist ein Partnerschaftsticket!
Bitte habe etwas Geduld, ein Teammitglied wird sich demn\xE4chst um deine Anfrage k\xFCmmern.`,
      color: 255
    });
    const embed2 = new EmbedBuilder({
      description: `**Server Beschreibung**
${description}`,
      fields: [
        { name: "Anzahl der Mitglieder", value: memberCount }
      ],
      color: 255
    });
    await channel.send({ content: "@everyone", embeds: [embed, embed2], components: [ticketButtons] }).then(async (message) => {
      await message.edit({ content: "" });
      interaction.reply({ content: `Dein Ticket wurde erstellt: ${channel}`, ephemeral: true });
    });
  }
};
export {
  partnership_default as default
};
