// contents/dist/perms-REXAQX2U.js
import {
  MemberManager
} from "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Admin/perms.ts
init_esm_shims();
import { SlashCommandBuilder, PermissionFlagsBits, EmbedBuilder, Colors } from "discord.js";
var command = {
  data: new SlashCommandBuilder().setName("perms").setDescription("Gibt die Berechtigungen eines Mitglieds wieder").addUserOption((input) => input.setName("user").setDescription("Welcher User?").setRequired(true)).addStringOption((input) => input.setName("search").setDescription("Suche nach einer Berechtigung")).setDefaultMemberPermissions(PermissionFlagsBits.Administrator),
  async execute(interaction, client) {
    const user = interaction.options.getMember("user");
    const guild = client.guild;
    const search = interaction.options.getString("search", false);
    const Member = new MemberManager(user, guild);
    const perms = Member.getPermissions();
    const embed = new EmbedBuilder({
      title: `Berechtigungen von ${user.user.username}`,
      description: `\`${perms.join("`, `")}\``,
      color: Colors.Gold,
      thumbnail: { url: Member.getAvatarUrl() || "" }
    });
    if (search) {
      const values = perms.filter((p) => p.toLowerCase().includes(search.toLowerCase()));
      embed.data.title += ` mit '${search.toLowerCase()}'`;
      if (values.length < 1) {
        embed.data.description = "```\nKeine Berechtigungen mit diesem Tag\n```";
      } else {
        embed.data.description = `\`${values.join("`, `")}\``;
      }
    }
    await interaction.reply({ embeds: [embed] });
  }
};
var perms_default = command;
export {
  perms_default as default
};
