// contents/dist/report-5V5JHHIR.js
import {
  ticketButtons
} from "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/Ticket/report.ts
init_esm_shims();
import { EmbedBuilder, ChannelType } from "discord.js";
var report_default = {
  id: "report",
  async execute(interaction) {
    const channel = await interaction.guild.channels.create({
      name: `ticket-${interaction.user.username}`,
      type: ChannelType.GuildText,
      parent: "1173314530521129042" /* ticket */,
      permissionOverwrites: [
        { id: interaction.user, allow: ["SendMessages", "ViewChannel"] },
        { id: interaction.guild.roles.everyone, deny: ["ViewChannel"] }
      ]
    });
    const embed1 = new EmbedBuilder({
      description: "Dies ist ein Report Ticket!\nBitte habe etwas Geduld, ein Teammitglied wird sich demn\xE4chst um deinen Report k\xFCmmern",
      color: 255
    });
    const embed2 = new EmbedBuilder({
      fields: [
        { name: "Gemeldeter User", value: interaction.fields.getTextInputValue("target") },
        { name: "Grund f\xFCr die Meldung", value: interaction.fields.getTextInputValue("reason") },
        { name: "Beweise?", value: interaction.fields.getTextInputValue("evidence") }
      ],
      color: 255
    });
    await channel.send({ content: "@everyone", embeds: [embed1, embed2], components: [ticketButtons] }).then(async (message) => {
      await message.edit({ content: "" });
      interaction.reply({ content: `Dein Ticket wurde erstellt: ${channel}`, ephemeral: true });
    });
  }
};
export {
  report_default as default
};
