// contents/dist/report-KBQZ47KT.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/General/report.ts
init_esm_shims();
import { EmbedBuilder, Colors, SlashCommandBuilder, PermissionFlagsBits } from "discord.js";
var command = {
  data: new SlashCommandBuilder().setName("report").setDescription("Melde einen User an das Serverteam").addUserOption((option) => option.setName("user").setDescription("Der zu meldende User").setRequired(true)).addStringOption((input) => input.setName("reason").setDescription("Grund f\xFCr die Meldeung").setRequired(true)).addAttachmentOption((input) => input.setName("image").setDescription("Erlaubte Dateien: png, jpg, jpeg, webp")).setDefaultMemberPermissions(PermissionFlagsBits.Administrator),
  async execute(interaction) {
    const user = interaction.options.getUser("user");
    const reason = interaction.options.getString("reason");
    const attachment = interaction.options.getAttachment("image");
    const embed = new EmbedBuilder({
      title: "Neuer Report",
      description: `${user} wurde von ${interaction.member} reportet. Grund f\xFCr den Report:
***${reason}***`,
      color: Colors.DarkOrange,
      timestamp: /* @__PURE__ */ new Date()
    });
    if (attachment) {
      if (attachment.name.endsWith(".png") || attachment.name.endsWith(".jpg") || attachment.name.endsWith(".jpeg") || attachment.name.endsWith(".webp")) {
        embed.setImage(attachment.url);
      }
    }
    await interaction.reply({ embeds: [embed] });
  }
};
var report_default = command;
export {
  report_default as default
};
