// contents/dist/roles-ELYRXFZM.js
import {
  MemberManager
} from "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Admin/roles.ts
init_esm_shims();
import { SlashCommandBuilder, PermissionFlagsBits, EmbedBuilder, Colors } from "discord.js";
var command = {
  data: new SlashCommandBuilder().setName("roles").setDescription("Gibt die Rollen eines MItglieds wieder").setDefaultMemberPermissions(PermissionFlagsBits.Administrator).addUserOption((input) => input.setName("user").setDescription("Welcher User?").setRequired(true)),
  async execute(interaction) {
    const user = interaction.options.getMember("user");
    const guild = interaction.guild;
    const Member = new MemberManager(user, guild);
    const roles = Member.getRoles();
    const embed = new EmbedBuilder({
      title: `${user.user.username} hat die folgenden Rollen:`,
      description: `${roles.join("\n")}`,
      color: Colors.Gold,
      thumbnail: { url: Member.getAvatarUrl() || "" }
    });
    interaction.reply({ embeds: [embed] });
  }
};
var roles_default = command;
export {
  roles_default as default
};
