// contents/dist/rules-NQMQZ7JY.js
import "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Admin/rules.ts
init_esm_shims();
import { SlashCommandBuilder, EmbedBuilder, ButtonBuilder, ActionRowBuilder, Colors } from "discord.js";
var command = {
  data: new SlashCommandBuilder().setName("rules").setDescription("Sende das aktualisierte Regelwerk").setDefaultMemberPermissions(0),
  async execute(interaction) {
    const channel = interaction.guild.channels.cache.get("1146113685962625128" /* rules */);
    const message = await channel.messages.fetch("1234592346402721803");
    const embed = new EmbedBuilder({
      title: "Unser Regelwerk",
      color: Colors.DarkGrey,
      fields: [
        {
          name: "\xA7A Grundlegende Bestimmungen",
          value: "A.1. Bei Betreten und Interaktion innerhalb \u300EAtlantis Lounge \u{1F30A}\u300F akzeptierst du unsere Richtlinien.\n\nA.2. Jede Person ist selbst f\xFCr die Integrit\xE4t ihres eigenen Zugangs verantwortlich.\n\u2800"
        },
        {
          name: "\xA7B Kontoverantwortlichkeit und Verhalten",
          value: "B.1. Benutzernamen oder Inhalt, der anderen schaden k\xF6nnte, sind unzul\xE4ssig.\n\nB.2. Dein Zugang sollte privat bleiben und nicht geteilt werden.\n\nB.3. Das Umgehen von verh\xE4ngten Einschr\xE4nkungen resultiert in strengeren Ma\xDFnahmen.\n\u2800"
        },
        {
          name: "\xA7C Umgang mit Bugs",
          value: "C.1. Das Nutzen oder Verbreiten von Bugs wird nicht geduldet.\n\nC.2. Jegliche Unregelm\xE4\xDFigkeiten sollten umgehend kommuniziert werden.\n\u2800"
        },
        {
          name: "\xA7D Interaktion zwischen Nutzern",
          value: "D.1. Transaktionen mit echtem Geld und automatisierte Interaktionen sind unzul\xE4ssig.\n\nD.2. Jegliche Form von Bedrohungen, besonders technischer Natur, ist strikt verboten.\n\nD.3. Jeder Nutzer sollte respektvoll und r\xFCcksichtsvoll gegen\xFCber anderen agieren.\n\u2800"
        },
        {
          name: "\xA7E Inhalte und geistiges Eigentum",
          value: "E.1. Inhalte, die auf dem Portal geteilt werden, m\xFCssen den allgemeinen Richtlinien entsprechen.\n\nE.2. Das Kopieren, Teilen oder anderweitige Verbreiten von Inhalten ohne Erlaubnis ist untersagt.\n\u2800"
        },
        {
          name: "\xA7F Schlussbestimmungen",
          value: "F.1. Verst\xF6\xDFe gegen diese Richtlinien k\xF6nnen zu Sanktionen f\xFChren.\n\nF.2.  \u300EAtlantis Lounge \u{1F30A}\u300F\nbeh\xE4lt sich das Recht vor, diese Richtlinien nach eigenem Ermessen zu \xE4ndern."
        }
      ]
    });
    const row = new ActionRowBuilder().addComponents([
      new ButtonBuilder({
        label: "Verifizieren",
        style: 3,
        customId: "verify"
      })
    ]);
    await message.edit({ embeds: [embed], components: [row] });
    interaction.reply({ content: "Regelwerk gesendet", ephemeral: true });
  }
};
var rules_default = command;
export {
  rules_default as default
};
