// contents/dist/selfrole-UKVOFCYG.js
import "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Admin/selfrole.ts
init_esm_shims();
import { SlashCommandBuilder, ButtonBuilder, EmbedBuilder, ActionRowBuilder, PermissionFlagsBits, Colors } from "discord.js";
var command = {
  data: new SlashCommandBuilder().setName("selfroles").setDescription("Sende die Selfroles in den selfrolechannel").setDefaultMemberPermissions(PermissionFlagsBits.Administrator),
  async execute(interaction, client) {
    await interaction.deferReply({ ephemeral: true });
    const channel = client.channels.cache.get("1148123116590071828" /* selfrole */);
    const genderEmbed = new EmbedBuilder({
      title: "Was bist du?",
      color: Colors.Gold
    });
    const ageEmbed = new EmbedBuilder({
      title: "Wie alt bist du?",
      color: Colors.Gold
    });
    const deviceEmbed = new EmbedBuilder({
      title: "Worauf spielst du?",
      color: Colors.Gold
    });
    const countryEmbed = new EmbedBuilder({
      title: "In welchem Land wohnst du?",
      color: Colors.Gold
    });
    const gamesEmbed = new EmbedBuilder({
      title: "Was spielst du?",
      color: Colors.Gold
    });
    const pingEmbed = new EmbedBuilder({
      title: "F\xFCr was m\xF6chtest du gepingt werden?",
      color: Colors.Gold
    });
    const row = ActionRowBuilder;
    const genderButtons = new row({
      components: [
        new ButtonBuilder({
          label: "M\xE4nnlich",
          style: 2,
          customId: "male",
          emoji: "\u{1F6B9}"
        }),
        new ButtonBuilder({
          label: "Weiblich",
          style: 2,
          customId: "female",
          emoji: "\u{1F6BA}"
        })
      ]
    });
    const ageButtons = new row({
      components: [
        new ButtonBuilder({
          label: "13+",
          style: 2,
          customId: "13+",
          emoji: "1\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "15+",
          style: 2,
          customId: "15+",
          emoji: "2\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "18+",
          style: 2,
          customId: "18+",
          emoji: "3\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "20+",
          style: 2,
          customId: "20+",
          emoji: "4\uFE0F\u20E3"
        })
      ]
    });
    const deviceButtons = new row({
      components: [
        new ButtonBuilder({
          label: "iOS",
          style: 2,
          customId: "ios",
          emoji: "\u{1F4F1}"
        }),
        new ButtonBuilder({
          label: "Android",
          style: 2,
          customId: "android",
          emoji: "\u{1F4F2}"
        }),
        new ButtonBuilder({
          label: "PC",
          style: 2,
          customId: "pc",
          emoji: "\u{1F5A5}\uFE0F"
        }),
        new ButtonBuilder({
          label: "Konsole",
          style: 2,
          customId: "konsole",
          emoji: "\u{1F579}\uFE0F"
        })
      ]
    });
    const countryButtons1 = new row({
      components: [
        new ButtonBuilder({
          label: "Deutschland",
          style: 2,
          customId: "germany",
          emoji: "\u{1F1E9}\u{1F1EA}"
        }),
        new ButtonBuilder({
          label: "\xD6sterreich",
          style: 2,
          customId: "\xF6sterreich",
          emoji: "\u{1F1E6}\u{1F1F9}"
        }),
        new ButtonBuilder({
          label: "Italien",
          style: 2,
          customId: "italy",
          emoji: "\u{1F1EE}\u{1F1F9}"
        }),
        new ButtonBuilder({
          label: "Niederlande",
          style: 2,
          customId: "netherlands",
          emoji: "\u{1F1F3}\u{1F1F1}"
        }),
        new ButtonBuilder({
          label: "Frankreich",
          style: 2,
          customId: "france",
          emoji: "\u{1F1EB}\u{1F1F7}"
        })
      ]
    });
    const countryButtons2 = new row({
      components: [
        new ButtonBuilder({
          label: "Schweiz",
          style: 2,
          customId: "switzerland",
          emoji: "\u{1F1E8}\u{1F1ED}"
        })
      ]
    });
    const gameButtons1 = new row({
      components: [
        new ButtonBuilder({
          label: "Minecraft Java",
          style: 2,
          customId: "mcjava",
          emoji: "1\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Minecraft Bedrock",
          style: 2,
          customId: "mcbedrock",
          emoji: "2\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Rocket league",
          style: 2,
          customId: "rocketleague",
          emoji: "3\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Fortnite",
          style: 2,
          customId: "fortnite",
          emoji: "4\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Clash of Clans",
          style: 2,
          customId: "coc",
          emoji: "5\uFE0F\u20E3"
        })
      ]
    });
    const gameButtons2 = new row({
      components: [
        new ButtonBuilder({
          label: "Valorant",
          style: 2,
          customId: "valorant",
          emoji: "1\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "GTA",
          style: 2,
          customId: "gta",
          emoji: "2\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Brawl Stars",
          style: 2,
          customId: "brawlstars",
          emoji: "3\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Call of Duty",
          style: 2,
          customId: "cod",
          emoji: "4\uFE0F\u20E3"
        })
      ]
    });
    const pingButtons1 = new row({
      components: [
        new ButtonBuilder({
          label: "Wahrheit oder Pflicht",
          style: 2,
          customId: "wop",
          emoji: "1\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Bumping",
          style: 2,
          customId: "bumping",
          emoji: "2\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Bot Status",
          style: 2,
          customId: "botstatus",
          emoji: "3\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Events",
          style: 2,
          customId: "events",
          emoji: "4\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Allgemeines",
          style: 2,
          customId: "general",
          emoji: "5\uFE0F\u20E3"
        })
      ]
    });
    const pingButtons2 = new row({
      components: [
        new ButtonBuilder({
          label: "Gewinnspiele",
          style: 2,
          customId: "giveaways",
          emoji: "1\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Neuigkeiten",
          style: 2,
          customId: "news",
          emoji: "2\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Umfragen",
          style: 2,
          customId: "polls",
          emoji: "3\uFE0F\u20E3"
        }),
        new ButtonBuilder({
          label: "Team \xC4nderungen",
          style: 2,
          customId: "staffchanges",
          emoji: "4\uFE0F\u20E3"
        })
      ]
    });
    await send(genderEmbed, [genderButtons]);
    await send(ageEmbed, [ageButtons]);
    await send(deviceEmbed, [deviceButtons]);
    await send(countryEmbed, [countryButtons1, countryButtons2]);
    await send(gamesEmbed, [gameButtons1, gameButtons2]);
    await send(pingEmbed, [pingButtons1, pingButtons2]);
    await interaction.editReply("Selfroles geschickt");
    async function send(embed, components) {
      await channel.send({ embeds: [embed], components });
    }
  }
};
var selfrole_default = command;
export {
  selfrole_default as default
};
