// contents/dist/support-MWYEEGSK.js
import {
  ticketButtons
} from "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/Ticket/support.ts
init_esm_shims();
import { EmbedBuilder, ChannelType } from "discord.js";
var support_default = {
  id: "support",
  async execute(interaction) {
    const channel = await interaction.guild.channels.create({
      name: `ticket-${interaction.user.username}`,
      type: ChannelType.GuildText,
      parent: "1173314530521129042" /* ticket */,
      permissionOverwrites: [
        { id: interaction.user, allow: ["SendMessages", "ViewChannel"] },
        { id: interaction.guild.roles.everyone, deny: ["ViewChannel"] },
        { id: "1156298949301379212" /* staff */, allow: ["SendMessages", "ViewChannel"] }
      ]
    });
    const embed1 = new EmbedBuilder({
      description: "Dies ist ein Support Ticket!\nBitte habe etwas Geduld, ein Teammitglied wird sich demn\xE4chst um dein Anliegen k\xFCmmern",
      color: 255
    });
    const embed2 = new EmbedBuilder({
      fields: [
        { name: "Anliegen", value: interaction.fields.getTextInputValue("issue") }
      ],
      color: 255
    });
    await channel.send({ content: "@everyone", embeds: [embed1, embed2], components: [ticketButtons] }).then(async (message) => {
      await message.edit({ content: "" });
      interaction.reply({ content: `Dein Ticket wurde erstellt: ${channel}`, ephemeral: true });
    });
  }
};
export {
  support_default as default
};
