// contents/dist/task-MZN5OEZQ.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Modals/Admin/task.ts
init_esm_shims();
import { EmbedBuilder, ButtonBuilder, ActionRowBuilder, Colors } from "discord.js";
var task_default = {
  id: "taskmodal",
  async execute(interaction) {
    const title = interaction.fields.getTextInputValue("tasktitle");
    const description = interaction.fields.getTextInputValue("taskdescription");
    const channel = interaction.guild.channels.cache.get("1200374840696246302");
    if (!channel) return interaction.reply({ content: "Der Dev Task Channel existiert nicht mehr. Bitte aktualisiere ihn im Code!", ephemeral: true });
    const finishButton = new ButtonBuilder({
      label: "Erledigt",
      customId: "taskfinished",
      style: 3
    });
    const claimButton = new ButtonBuilder({
      label: "\xDCbernehmen",
      customId: "taskclaim",
      style: 1
    });
    const row = new ActionRowBuilder().addComponents([finishButton, claimButton]);
    const embed = new EmbedBuilder({
      title,
      description,
      timestamp: /* @__PURE__ */ new Date(),
      color: Colors.DarkPurple
    });
    const message = await channel.send({ content: "<@&1146117778483450048>", embeds: [embed], components: [row] });
    await message.pin();
    interaction.reply({ content: "Die Aufgabe wurde erfolgreich \xFCbermittelt", ephemeral: true });
  }
};
export {
  task_default as default
};
