// contents/dist/test-ML74SO3G.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Admin/test.ts
init_esm_shims();
import { SlashCommandBuilder, EmbedBuilder, AttachmentBuilder, Colors } from "discord.js";
import { profileImage } from "discord-arts";
var command = {
  data: new SlashCommandBuilder().setName("canvas-test").setDescription("Canvas Tests").setDefaultMemberPermissions(0),
  async execute(interaction, client) {
    await interaction.deferReply();
    const member = interaction.member;
    const buffer = await profileImage(member.user.id, {
      customDate: (/* @__PURE__ */ new Date()).toLocaleDateString("ru"),
      badgesFrame: true,
      customTag: "Willkommen",
      presenceStatus: member.presence?.status
    }).catch(console.log);
    const embed = new EmbedBuilder({
      title: "Willkommen",
      description: `Hallo ${member} und ein herzliches Willkommen auf unserem Server! \u{1F44B}
Wir freuen uns, dass du unserer Community beitreten bist \u2764\uFE0F

\u{1F91D} Falls du Lust hast, kannst du dich in <#1182194022010322975> unserer Community vorstellen - teile uns gerne etwas \xFCber dich, deine Interessen und was dich hierhin f\xFChrt mit.

\u{1F4DC} Bevor du loslegen kannst, nimm dir bitte einen Moment Zeit, um unsere Serverregeln durchzulesen. Wir m\xF6chten, dass jeder hier eine tolle Zeit hat, und die Einhaltung der Richtlinien tr\xE4gt dazu bei, ein positives Umfeld f\xFCr alle zu schaffen.

Wir freuen uns dass du unserem Server beigetreten bist. \u{1F389}

Wir sind jetzt **${(await member.guild.members.fetch()).size} Mitglieder**

Mit freundlichen Gr\xFC\xDFen:
**Das Server-Team**`,
      color: Colors.DarkGrey,
      fields: [
        {
          name: "Wichtige Channel",
          value: "- Unsere Regeln in <#1146113685962625128> \n- Selfroles in <#1148123116590071828> \n- Support in <#1153007107688386752>"
        }
      ]
    });
    if (buffer instanceof Buffer) {
      const attachment = new AttachmentBuilder(buffer, { name: "image.png" });
      embed.setImage("attachment://image.png");
      await interaction.editReply({ content: `${member}`, embeds: [embed], files: [attachment] });
    } else {
      await interaction.editReply({ content: `${member}`, embeds: [embed] });
    }
  }
};
var test_default = command;
export {
  test_default as default
};
