// contents/dist/unban-request-CRH6ZD3A.js
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Buttons/Security/unban-request.ts
init_esm_shims();
import { ActionRowBuilder, ModalBuilder, TextInputBuilder } from "discord.js";
var unban_request_default = {
  id: "unban-request",
  async execute(interaction) {
    const modal = new ModalBuilder({
      customId: "unban-request",
      title: "Entbannungsantrag",
      components: [
        new ActionRowBuilder().addComponents([
          new TextInputBuilder({
            customId: "ban-reason",
            label: "Warum wurdest du gebannt?",
            style: 1,
            required: true
          })
        ]),
        new ActionRowBuilder().addComponents([
          new TextInputBuilder({
            customId: "unban-reason",
            label: "Warum m\xF6chtest du entbannt werden?",
            style: 2,
            required: true,
            placeholder: "Bitte schreibe ganze S\xE4tze"
          })
        ])
      ]
    });
    await interaction.showModal(modal);
  }
};
export {
  unban_request_default as default
};
