// contents/dist/warn-GH27KCTN.js
import {
  warns_default
} from "./chunk-OPC27YAZ.js";
import "./chunk-P5JGOPVX.js";
import {
  init_esm_shims
} from "./chunk-WWCIODDN.js";

// Commands/Moderation/warn.ts
init_esm_shims();
import { SlashCommandBuilder, EmbedBuilder, PermissionFlagsBits, Colors } from "discord.js";
var command = {
  data: new SlashCommandBuilder().setName("warn").setDescription("Warne einen User").addUserOption((input) => input.setName("user").setDescription("Der User der gewarnt werden soll").setRequired(true)).addStringOption((input) => input.setName("reason").setDescription("Der Grund f\xFCr den Warn").setRequired(true)).setDefaultMemberPermissions(PermissionFlagsBits.ModerateMembers),
  async execute(interaction) {
    const reason = interaction.options.get("reason", true).value;
    const member = interaction.options.getMember("user");
    const channel = interaction.guild.channels.cache.get("1160607902210470009" /* warn */);
    await warns_default.findOne({ userId: member.user.id }).then(async (User) => {
      if (User) {
        User.warns.push({
          date: (/* @__PURE__ */ new Date()).toLocaleDateString("ru"),
          id: generateWarnId(),
          moderator: interaction.user.username,
          reason
        });
        await User.save();
      } else {
        await warns_default.create({
          userId: member.user.id,
          warns: [
            {
              date: (/* @__PURE__ */ new Date()).toLocaleDateString("ru"),
              id: generateWarnId(),
              moderator: interaction.user.username,
              reason
            }
          ]
        });
      }
    });
    const embed = new EmbedBuilder({
      title: "User gewarnt",
      description: `<a:redlight:1211374559224135700> ${member} wurde von ${interaction.user.username} gewarnt
Grund: **${reason}**`,
      color: Colors.Red
    });
    const dmEmbed = new EmbedBuilder({
      author: { name: interaction.guild.name, iconURL: interaction.guild.iconURL() || "" },
      title: "Du wurdest gewarnt",
      fields: [
        { name: "Moderator", value: `${interaction.user} (${interaction.user.username})` },
        { name: "Grund", value: reason }
      ]
    });
    await channel.send({ embeds: [embed] });
    interaction.reply({ content: `Du hast ${member} gewarnt!
Grund: **${reason}**`, ephemeral: true });
    await member.send({ embeds: [dmEmbed] }).catch(console.log);
  }
};
var warn_default = command;
function generateWarnId() {
  const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUFWXYZ1234567890";
  let id = "";
  for (let index = 0; index < 5; index++) {
    const randIndex = Math.floor(Math.random() * charset.length);
    id += charset.charAt(randIndex);
  }
  return id;
}
export {
  warn_default as default
};
