// contents/dist/Enums/Selfroles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Selfroles = void 0;
var Selfroles;
(function (Selfroles) {
    Selfroles["male"] = "1165352617191407656";
    Selfroles["female"] = "1165352742953439403";
    Selfroles["age13"] = "1146341429153640448";
    Selfroles["age15"] = "1148599932526264320";
    Selfroles["age18"] = "1148600095495950376";
    Selfroles["age20"] = "1148600180124430376";
    Selfroles["ios"] = "1148600312840597614";
    Selfroles["android"] = "1148600380561829921";
    Selfroles["pc"] = "1148600431359049789";
    Selfroles["konsole"] = "1148600598804041838";
    Selfroles["deutschland"] = "1149569712418734080";
    Selfroles["schweiz"] = "1149569751824220220";
    Selfroles["\u00F6sterreich"] = "1149569815229497444";
    Selfroles["italien"] = "1148637509962694747";
    Selfroles["niederlande"] = "1175444444129017897";
    Selfroles["frankreich"] = "1148637572516556840";
    Selfroles["mcjava"] = "1148667098072096788";
    Selfroles["mcbedrock"] = "1146113684486225967";
    Selfroles["rocketleague"] = "1148667193962283008";
    Selfroles["fortnite"] = "1148667318600220782";
    Selfroles["valorant"] = "1148667826245210132";
    Selfroles["clashofclans"] = "1148638144384749750";
    Selfroles["gta"] = "1148637979405996084";
    Selfroles["brawlstars"] = "1148638085379276810";
    Selfroles["callofduty"] = "1148638042014355496";
    Selfroles["poll"] = "1148638225963950220";
    Selfroles["news"] = "1148638271493132410";
    Selfroles["giveaways"] = "1148638318263799869";
    Selfroles["wop"] = "1175441946852991086";
    Selfroles["bump"] = "1173663419531022447";
    Selfroles["botstatus"] = "1160253015639457842";
    Selfroles["event"] = "1149728263179091989";
    Selfroles["allgemeines"] = "1149728389809311776";
    Selfroles["staffchanges"] = "1148638515840700497";
})(Selfroles || (exports.Selfroles = Selfroles = {}));
